pkg/handler/collector/git/git_test.go

```diff
@@ -0,0 +1,170 @@
+//
+// Copyright 2022 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package git_collector
+
+import (
+	"context"
+	"io"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/guacsec/guac/pkg/handler/processor"
+	"github.com/guacsec/guac/pkg/logging"
+)
+
+func Test_gitCol_RetrieveArtifacts(t *testing.T) {
+	ctx := logging.WithLogger(context.Background())
+	logger := logging.FromContext(ctx)
+	type fields struct {
+		url      string
+		dir      string
+		poll     bool
+		interval time.Duration
+	}
+	tests := []struct {
+		name                   string
+		fields                 fields
+		preCreateDir           bool
+		wantDirEmpty           bool
+		wantErr                bool
+		numberOfFilesCollected int
+	}{{
+		name: "get repo",
+		fields: fields{
+			url:      "https://github.com/shafeeshafee/test-123.git",
+			dir:      "./guac-data-test",
+			poll:     false,
+			interval: time.Millisecond,
+		},
+		preCreateDir:           false,
+		wantDirEmpty:           false,
+		wantErr:                false,
+		numberOfFilesCollected: 9,
+	}, {
+		name: "if repo exist",
+		fields: fields{
+			url:      "https://github.com/shafeeshafee/test-123.git",
+			dir:      "./guac-data-test",
+			poll:     false,
+			interval: time.Millisecond,
+		},
+		preCreateDir:           true,
+		wantDirEmpty:           false,
```

Comment:
clarifying question: what's the difference between `numberOfFilesCollected` and `wantDirEmpty` for the 0 case?

Reply:
from what I remember with my convo with @pxp928, they serve different purposes, `numberOfFilesCollected` is used to check that the expected number of files have been collected - 9 being the total number of files it expects while `wantDirEmpty` is set to ensure whether we want the collection process to start from a clean state before the test case.